app/views/internal/feedback_messages/_abuse_reports.html.erb

```diff
@@ -44,26 +44,61 @@
             <span>
               <% if params[:status] == "Open" || params[:status].blank? %>
                 <% if reaction.reactable_type == "User" %>
-                  <%= form_for [:internal, reaction], html: { class: "d-inline" } do |f| %>
-                    <%= f.hidden_field :status, value: "confirmed" %>
-                    <%= f.submit "CONFIRMED", class: "btn btn-success btn-sm", data: { confirm: "Are you sure?" } %>
-                  <% end %>
+                  <button class="btn btn-success btn-sm js-reaction-confirm" type="button" data-reactable="user" data-id="<%= reaction.id %>" data-status="confirmed">CONFIRM</button>
                 <% else %>
-                  <%= form_for [:internal, reaction], html: { class: "d-inline" } do |f| %>
-                    <%= f.hidden_field :status, value: "confirmed" %>
-                    <%= f.submit "CONFIRMED", class: "btn btn-success btn-sm" %>
-                  <% end %>
-                <% end %>
-                <%= form_for [:internal, reaction], html: { class: "d-inline" } do |f| %>
-                  <%= f.hidden_field :status, value: "invalid" %>
-                  <%= f.submit "INVALID", class: "btn btn-danger btn-sm" %>
+                  <button class="btn btn-success btn-sm js-reaction-confirm" type="button" data-reactable="not-user" data-id="<%= reaction.id %>" data-status="confirmed">CONFIRM</button>
                 <% end %>
+                <button class="btn btn-danger btn-sm js-reaction-confirm" type="button" data-id="<%= reaction.id %>" data-status="invalid">INVALID</button>
               <% end %>
             </span>
           </div>
-          <hr>
+          <hr id="js__reaction__div__hr__<%= reaction.id %>">
         <% end %>
       </div>
     </div>
   </div>
 </div>
+
+<script>
+  function updateReactionStatus(id, status){
+    fetch(`/internal/reactions/${id}`, {
+     method: 'PATCH',
+     headers: {
+      Accept: 'application/json',
+       'X-CSRF-Token': document.querySelector("meta[name='csrf-token']").content,
+       'Content-Type': 'application/json',
+     },
+     body: JSON.stringify({
+       id: id,
+       status: status,
+      }),
+     credentials: 'same-origin'
+   })
+     .then(response => response.json()
+                                     .then(json => {
```

Comment:
maybe we can split this in two callbacks, so that it doesn't indent this much.

```js
    .then(response => response.json())
    .then(json => {
```

something like this :)

Reply:
👍